### PR TITLE
Update flyte auth documentation

### PR DESCRIPTION
**Pull request by jcourteau**
This updates the authentication docs to deal with a few challenges I faced when setting up auth:
* fixes incorrect indentation in a code example
* adds more explicit instructions for an Okta integration
* adds some warnings, clarifications, and callouts around problem areas - such as audience vs authorizedUri mismatches, and missing scopes.

### CODE REVIEW

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 1]
```diff
@@ -50,6 +50,7 @@ Prerequisites
 =============
 
 The following is required for non-sandbox deployments:
+
```
Comment (jcourteau) [inline rsts/deployment/cluster_config/auth_setup.rst:53]: this should fix a rendering issue with the bullets in the original docs

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 2]
```diff
@@ -228,12 +233,30 @@ To set up an external OAuth2 Authorization Server, follow the instructions below
...
 
-   1. Under security -> API, click `Add Authorization Server`. Set the audience to the public URL of FlyteAdmin (e.g. https://flyte.mycompany.io/).
-   2. Under `Access Policies`, click `Add New Access Policy` and walk through the wizard to allow access to the authorization server.
-   3. Under `Scopes`, click `Add Scope`. Set the name to `all` (required) and check `Require user consent for this scope` (recommended).
-   4. Create 2 apps (for Flytectl and Flytepropeller) to enable these clients to communicate with the service.
-      Flytectl should be created as a `native client`.
-      Flytepropeller should be created as an `OAuth Service` and note the client ID and client Secrets provided.
+   Okta's custom authorization servers are available through an add-on license. The free developer accounts do include access, which you can use to test before rolling out the configuration more broadly.
+
+   1. Under security -> API, click `Add Authorization Server`. Set the audience to the public URL of FlyteAdmin (e.g. https://example.foobar.com).
```
Comment (jcourteau) [inline rsts/deployment/cluster_config/auth_setup.rst:238]: I have updated the example public URLs throughout this doc to `https://example.foobar.com` to match the first given example

File: rsts/deployment/cluster_config/auth_setup.rst [hunk 3]
```diff
@@ -267,18 +290,18 @@ Apply Configuration
...
                     #metadataUrl: .well-known/openid-configuration #Uncomment for keycloak
 
-            thirdPartyConfig:
-                flyteClient:
-                    # 3. Replace with a new Native/Public Client ID provisioned in the custom authorization server.
-                    clientId: flytectl
+                thirdPartyConfig:
+                    flyteClient:
+                        # 3. Replace with a new Native/Public Client ID provisioned in the custom authorization server.
+                        clientId: flytectl
```
Comment (jcourteau) [inline rsts/deployment/cluster_config/auth_setup.rst:296]: The `thirdPartyConfig` is supposed to be an element of the `appAuth` object, not on the same level.

**Review by EngHabu — DISMISSED** [review]

**Comment by jcourteau** [comment]
> Looks like rendering here isn't working great:

I think the `tabbed::` container is doing something odd with nested lists. I'll flatten the lists, or split them up a bit better instead.

**Comment by jcourteau** [comment]
https://flyte--2611.org.readthedocs.build/en/2611/deployment/cluster_config/auth_setup.html  - formatting issues look resolved.

**Comment by EngHabu** [comment]
I think the only remaining issue is DCO test is failing. It looks like you can run (**on your fork**):
```bash
git rebase HEAD~5 --signoff
git push --force-with-lease origin master
```

**Comment by EngHabu** [comment]
Spoke too soon, @jcourteau , looks like another test has failed. https://github.com/flyteorg/flyte/runs/7011204865?check_suite_focus=true

**Comment by jcourteau** [comment]
Well that seems to have worked. Cleaned out the errant merge commit from the PR which was messing with DCO; let me know if there's anything missing.

**Review by EngHabu — APPROVED** [review]

**Comment by EngHabu** [comment]
Awesome! Thanks again

**Pull request merged**